File: truth_set/python/structure/organisation/dead-code/OrderProcessor.py
from dataclasses import dataclass
from typing import List
from os import path

def calculate_discount(amount: float) -> float:
    discount_rate = 0.1
    return amount * discount_rate

@dataclass
class Order:
    id: int
    total: float

class OrderProcessor:
    def __init__(self, logger):
        self.logger = logger
        self.orders: List[Order] = []
        self.tax_rate = 0.07
        self.error_message = "Order was not processed"

    def process_order(self, order: Order) -> None:
        self.orders.append(order)
        self.logger.log(f"Processing order ID: {order.id}")
        
        if True:
            self.apply_tax(order)
        else:
            self.apply_discount(order)
            
        if False:
            self.send_confirmation_email(order)
            
        processing_time = self.calculate_processing_time(order)
        self.logger.log(f"Processing time for order ID {order.id}: {processing_time}ms")

    def apply_tax(self, order: Order) -> None:
        order.total += order.total * self.tax_rate
        self.logger.log(f"Applied tax to order ID: {order.id}")

    def apply_discount(self, order: Order) -> None:
        discount = calculate_discount(order.total)
        order.total -= discount
        self.logger.log(f"Applied discount to order ID: {order.id}")

    def send_confirmation_email(self, order: Order) -> None:
        self.logger.log(f"Sent confirmation email for order ID: {order.id}")

    def calculate_processing_time(self, order: Order) -> int:
        return 100

class Logger:
    def log(self, message: str) -> None:
        print(message)

logger = Logger()
processor = OrderProcessor(logger)
new_order = Order(id=1, total=200)
processor.process_order(new_order)

File: truth_set/javascript/structure/organisation/dead-code/OrderProcessor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function calculateDiscount(amount) {
    var discountRate = 0.1;
    return amount * discountRate;
}
var OrderProcessor = /** @class */ (function () {
    function OrderProcessor(logger) {
        this.logger = logger;
        this.orders = [];
        this.taxRate = 0.07;
        this.errorMessage = "Order was not processed";
    }
    OrderProcessor.prototype.processOrder = function (order) {
        this.orders.push(order);
        this.logger.log("Processing order ID: ".concat(order.id));
        if (true) {
            this.applyTax(order);
        }
        else {
            this.applyDiscount(order);
        }
        if (false) {
            this.sendConfirmationEmail(order);
        }
        var processingTime = this.calculateProcessingTime(order);
        this.logger.log("Processing time for order ID ".concat(order.id, ": ").concat(processingTime, "ms"));
    };
    OrderProcessor.prototype.applyTax = function (order) {
        order.total += order.total * this.taxRate;
        this.logger.log("Applied tax to order ID: ".concat(order.id));
    };
    OrderProcessor.prototype.applyDiscount = function (order) {
        var discount = calculateDiscount(order.total);
        order.total -= discount;
        this.logger.log("Applied discount to order ID: ".concat(order.id));
    };
    OrderProcessor.prototype.sendConfirmationEmail = function (order) {
        this.logger.log("Sent confirmation email for order ID: ".concat(order.id));
    };
    OrderProcessor.prototype.calculateProcessingTime = function (order) {
        return 100;
    };
    return OrderProcessor;
}());
var Logger = /** @class */ (function () {
    function Logger() {
    }
    Logger.prototype.log = function (message) {
        console.log(message);
    };
    return Logger;
}());
var logger = new Logger();
var processor = new OrderProcessor(logger);
var newOrder = { id: 1, total: 200 };
processor.processOrder(newOrder);